rust/agama-dbus-server/src/questions.rs

```diff
@@ -227,10 +242,35 @@ impl Questions {
         Ok(())
     }
 
-    /// sets questions to be answered by default answer instead of asking user
-    async fn use_default_answer(&mut self) -> Result<(), Error> {
-        log::info!("Answer questions with default option");
-        self.answer_strategies.push(Box::new(DefaultAnswers {}));
+    /// property that defines if questions is interactive or automatically answered with
+    /// default answer
+    #[dbus_interface(property)]
+    fn interactive(&self) -> bool {
+        let last = self.answer_strategies.last();
+        if let Some(real_strategy) = last {
+            real_strategy.id() != DefaultAnswers::id()
+        } else {
+            true
+        }
+    }
+
+    #[dbus_interface(property)]
+    fn set_interactive(&mut self, value: bool) {
+        log::info!("set interactive to {}", value);
+        if value {
```

Comment:
Implementation nitpicking: I would find it easier to understand like this, and the pattern is better generalized to more-valued properties:

```rs
if value == self.interactive() {
  log::info!("Interactive property unchanged, {}", value);
  return;
}
log::info!("set interactive to {}", value);

if value {
  ...pop...
} else {
  ...push...
}
```